moztelemetry/dataset.py

```diff
@@ -40,6 +41,30 @@ def _group_by_size_greedy(obj_list, tot_groups):
         current_group.append(obj)
     return groups
 
+def _group_by_equal_size(obj_list, tot_groups, threshold=pow(2, 32)):
+    sub_list = []
+    groups = [[]]
+    pq = []
+    tmp_object = none
+
+    if tot_groups <= 1:
+        group_by_size_greedy(obj_list, tot_groups)
+
+    pq = heapq.heapify(obj_list)
```

Comment:
Items in the priority queue should be tuples of `(priority, object)` as per the [second example](https://docs.python.org/2/library/heapq.html#basic-examples) in the heapq docs.

Reply:
I use the heapify() method to make the heap, found [here](https://docs.python.org/2/library/heapq.html#heapq.heapify). It takes a list and sorts it in place. I believe the default behavior is to prioritize by smallest.

Reply:
I think heapify needs a comparison operator like `.__cmp__` or `.__lt__` to be defined. Dictionaries are like hash-tables, and don't have a natural order like integers or strings do.